tests/unit_tests/test_matchmaker_queue.py

```diff
@@ -224,7 +225,83 @@ async def test_queue_pop_time_moving_average_size():
     assert t1.time_until_next_pop(0, 100) == config.QUEUE_POP_TIME_MAX
 
 
+@given(rating=st.integers())
+def test_queue_map_pools_empty(queue_factory, rating):
+    queue = queue_factory()
+    assert queue.get_map_pool_for_rating(rating) is None
+
+
+@given(rating=st.integers())
+def test_queue_map_pools_any_range(queue_factory, rating):
+    queue = queue_factory()
+    map_pool = MapPool(0, "pool")
+    queue.add_map_pool(map_pool, None, None)
+
+    assert queue.get_map_pool_for_rating(rating) is map_pool
+
+
+@given(rating=st.integers(), low=st.integers())
+def test_queue_map_pools_lower_bound(queue_factory, rating, low):
+    queue = queue_factory()
+    map_pool = MapPool(0, "pool")
+    queue.add_map_pool(map_pool, low, None)
+
+    if rating < low:
+        assert queue.get_map_pool_for_rating(rating) is None
+    else:
+        assert queue.get_map_pool_for_rating(rating) is map_pool
+
+
+@given(rating=st.integers(), high=st.integers())
+def test_queue_map_pools_upper_bound(queue_factory, rating, high):
+    queue = queue_factory()
+    map_pool = MapPool(0, "pool")
+    queue.add_map_pool(map_pool, None, high)
+
+    if rating > high:
+        assert queue.get_map_pool_for_rating(rating) is None
+    else:
+        assert queue.get_map_pool_for_rating(rating) is map_pool
+
+
+@given(rating=st.integers(), low=st.integers(), high=st.integers())
+def test_queue_map_pools_bound(queue_factory, rating, low, high):
+    queue = queue_factory()
+    map_pool = MapPool(0, "pool")
+    queue.add_map_pool(map_pool, low, high)
+
+    if low <= rating <= high:
+        assert queue.get_map_pool_for_rating(rating) is map_pool
+    else:
+        assert queue.get_map_pool_for_rating(rating) is None
+
+
+@given(
+    rating=st.integers(),
+    low1=st.integers(),
+    high1=st.integers(),
+    low2=st.integers(),
+    high2=st.integers()
+)
+def test_queue_multiple_map_pools(
+    queue_factory, rating, low1, high1, low2, high2
+):
+    queue = queue_factory()
+    map_pool1 = MapPool(0, "pool1")
+    map_pool2 = MapPool(1, "pool2")
+    queue.add_map_pool(map_pool1, low1, high1)
+    queue.add_map_pool(map_pool2, low2, high2)
+
+    if low1 <= rating <= high1:
+        assert queue.get_map_pool_for_rating(rating) is map_pool1
+    elif low2 <= rating <= high2:
+        assert queue.get_map_pool_for_rating(rating) is map_pool2
```

Comment:
I think this test relies on `queue` saving the `MapPool`s in  order added and returning the first pool that was added if there is ambiguity.
That's not a bad thing, but maybe something to keep in mind - this test might break if `queue` ever chooses to store map pools in e.g. a dict and then we might not have wanted to document this assumption.
I'm ok with it as is but you could consider
```python3
eligible_map_pools = [map_pools[i] for i in range(2) if low[i] <= rating <= high[i]]
if eligible_map_pools:
  assert queue.get_map_pool_for_rating(rating) in eligible_map_pools
else:
  assert queue.get_map_pool_for_rating(rating) is None
```